### PR TITLE
Hide network section in sidebar when no drives are detected

### LINKED ISSUE #3924: Enhancement: Hide network section in the sidebar when no network drives are detected

### PR DESCRIPTION
**Pull request by Daninator1**
Closes #3924 

I just added an additional check before creating the network drive LocationItem.

This is my first PR, feedback is appreciated 🙂

### CODE REVIEW

**Comment by XPoppyX** [comment]
Still visible to me
Captura de pantalla 2021-03-16 190358

**Comment by Daninator1** [comment]
Oh, seems like I missed that, will alter the check tomorrow probably, thanks!

File: Files/Filesystem/NetworkDrivesManager.cs [hunk 1]
```diff
@@ -112,7 +112,7 @@ private async Task SyncSideBarItemsUI()
                     MainPage.SideBarItems.BeginBulkOperation();
 
                     var section = MainPage.SideBarItems.FirstOrDefault(x => x.Text == "SidebarNetworkDrives".GetLocalized()) as LocationItem;
-                    if (section == null)
+                    if (section == null && this.drivesList.Count > 0)
```
Comment (yaichenbaum) [inline Files/Filesystem/NetworkDrivesManager.cs:115]: I don't think this will work, not all mapped drives will have a letter.

**Comment by Daninator1** [comment]
Alright, so I adapted the check to include the network folder which gets added to the list before. As I had no way to tell if the DriveItem is in fact just the network folder, I added a custom device id to perform the check. The other solution would have been to go strictly by the count of the items (Count > 1), although that wouldn't have been as flexible for future changes. Any thoughts on that?

File: Files/Filesystem/NetworkDrivesManager.cs (hunk 1 above)
Comment (Daninator1) [reply Files/Filesystem/NetworkDrivesManager.cs]: I'm not sure if I get your point about drive letters, any detected network drives should be in that list (+ network folder), so my thought was that as long as there are no drives in that list (apart from the network folder itself) we do not create the section in the sidebar. Or did I get something wrong?

**Review by gave92 — APPROVED** [review]

**Pull request merged**